src/manpages/airosconf.1
.TH AIROSCONF 1 "May  4, 2008"

.SH NAME
airosconf - update airoscript and airosperl config

.SH DESCRIPTION
airosconf is a tool designed for automating update/editing of airoscript and
airosperl config files.

.br.

.SH SYNOPSIS
.B airosconf 
[options]

.SH OPTIONS
.TP
--set
  --to-airoscript
  --to-airosperl

--update
  --from-airoscript
  --from-airosperl

.SH AUTHOR
Airosconf was written by David Francos Cuartero (XayOn) <yo.orco@gmail.com>.
.br
This manual page was written by David Francos Cuartero (XayOn) <yo.orco@gmail.com>.

.SH SEE ALSO
.br
.B airdecap-ng(1)
.br
.B airdriver-ng(1)
.br
.B aireplay-ng(1)
.br
.B airmon-ng(1)
.br
.B airodump-ng(1)
.br
.B airolib-ng(1)
.br
.B airsev-ng(1)
.br
.B airtun-ng(1)
.br
.B buddy-ng(1)
.br
.B easside-ng(1)
.br
.B ivstools(1)
.br
.B kstats(1)
.br
.B makeivs-ng(1)
.br
.B packetforge-ng(1)
.br
.B wesside-ng(1)
.br
.B aircrack-ng(1)


CHANGELOG
Changelog for 2.0.9
-fixed client and clientless attacks chopchop and frag packet injection not generating IVs because -h MAC was not set
-force frag, chopchop and aircrack-ng xterms to stay on screen after successfull command so you can see results
-cleaned the code some more
-cleaning functions needs more tweaks
-atheros users need to use options 9 (monitor mode) before other commands to create-destroys correct VAPS
-fixed "unary operator expected" msg (harmless bug) (not fully fixed because of "" generating errors in logic tests)
-tweaked 2 steps attacks (frag, chopchop) to go to menu to inject after successfull first step
-added option to jump to injection menu (added this to the main menu)
-fixed AP encryption detection for cracking step (bug was introduced by "")
-lowered adapter speed to 1M for wesside-ng attacks
-added -hold to mdk attack to allow users to see results before the windows closes

Changelog for 2.0.8 (27/01/2008)
-added wesside-ng airoscript style usage
-added interface type detection
-tweaked menu
-first attempt at making airoscript work with different adapters (add more card support)(BIG THX TO GREEN-FREQ!!!)
-atheros device are now supported
-added mdk3
-added back to main menu options
-fixed chopchop attack (BIG THX TO GREEN-FREQ!!!)
-tweaked FAKE_MAC function for atheros

Changelog for 2.0.7 BETA1 (21/01/2007)
-changed capture from IVs to full date (ivs to cap)
-updated script for new implementation of fragmentation attack
-other minor modifications and tweaks

Changelog for PRE-1.8.2-BETA1 (30/11/2006)
-fixed all commands leading to : "head: Binary: invalid number of lines" (MAJOR THX TO ANDREA)
-fixed all commands where SSID was truncated (MAJOR THX TO PILOTSNIPES)

Changelog for 1.8.1 (03/11/2006)
-bug fix release
-fixed some problem in the SSID detection function (MAJOR THX TO andreahaku)
-cleaned a few more typos and bugs
-documentation completed a bit

Changelog for 1.8 (17/10/2006)
-dropped support for aircrack-ng 0.6.1 and older
-will only work with patched version of aircrack-ng 0.6.2 (patches can be found in the forum under the programming section)
-specifique channels hopping (for example channel 1,6,11 or 1-3,5-7,9,11)
-Access point filtering during attack (airodump will only list attacked Access point)
-new SSID detection, works with ALL ssid (supports all character and spacing possible)
-new Access point selection menu that list all AP with encryption type (eliminates the WEP? problem)
-option 2) now get you directly to a full list of AP
-chopchop attack and fragmentation attack implement
-WPA dictionnary attack implemented
-changed menu 3,4,5 based on detection of AP encryption
-added option to jump to associated client list from ap selection
-color change support for xterm
-Windows size and placement fully changeable from variables settings
-changed filenames from SSID to MAC because of possible issues if weird SSID (to eliminate security menace (if SSID is something like "rm -rf /*" etc...)
-added option an option to run "1) scan 2) select 3) attack" one after the other
-added some comments in the script to help users
-added greetings message (can be disabled setting WELCOME="1" to "0")

Changelog for 1.7RC7 (24/09/2006)
-added some resilience to the configure function now works properly with hidden ssid's and rausb devices 

Changelog for 1.7RC6 (19/09/2006)
-Code cleanup
-added deauth command to solointeractiveattack function
-changed base variables (fudge factor & inject rate)

Changelog for 1.7RC5 (14/09/2006)
-Fixed minor cosmetic issue caused by carriage return char when ssid is missing (15/9/06)
-Fixed the weird character match issue!! edit at will!!!
-added an abitrary variable ($acv) to the aircrackversioncheck function which can be used in future to keep the script backwards compatible (had to guess at what version 0.6.0 of aircrack-ng displays when run on it's own? and version 0.5.so might not work??)
-used new variable in crack function to check version and run correct crack command for the aricrack version installed fixing the bug on the B|T forum.

Changelog for 1.7RC4 (13/09/2006)
-corrected minor speellling issues ;o) 
-added aircrack-ng version detection function currently all it does is display aircrack version in the menu function (should be useful when new functionality is added to aircrack to allow backwards compatibility)
-added solointeractiveattack function (designed to run just the airreplay interactive attack function only for use with an attack that has stalled i.e. data has stopped increasing, the user can try this for another injection thread! (might not actually work?? needs more testing does spawn and seems to work!)
-added option to manually add ssid if a blank ssid is detected (this has some problems when certain editors are used as this is not strictly blank but a non printable character recommend kwrite on Back|track to edit script)


src/tools/airosconf.sh
#!/bin/bash
# Airoscript configuration management script
# Same licensing as the rest of the airoscript project (gpl2+)
# Warning, this is a very limited way to configure airoscript, use it at your own risk.
# Anyway, it's usefull for new installations.
# Copyright (C) 2009 David Francos Cuartero
#        This program is free software; you can redistribute it and/or
#        modify it under the terms of the GNU General Public License
#        as published by the Free Software Foundation; either version 2
#        of the License, or (at your option) any later version.

#        This program is distributed in the hope that it will be useful,
#        but WITHOUT ANY WARRANTY; without even the implied warranty of
#        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#        GNU General Public License for more details.

#        You should have received a copy of the GNU General Public License
#        Along with this program; if not, write to the Free Software
#        Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.


function _set_airoscript(){
	$tmp=`mktemp`
	grep -v $third /etc/airoscript.conf > $tmp
	echo "$third = \"$fourth\" " > /etc/airoscript.conf
}

function _set_airosperl(){
	if [ $a != "--force-yes" ]; then
		read -p "Are you sure to do this? If you've got an improper (more than one scalar per line) conffile, this can break it. (N/y)" YN && [[ $YN != "y" ]] && exit 
	fi
	if [ "`perl -e 'print ref($3)' `" == "SCALAR" ] || [ "`perl -e 'print ref ($3)'`" == "" ];
	then
		tmp=`mktemp`
		grep -v "$third" /etc/airosperl.conf |grep -v "1;"> $tmp
		echo -e "\nour \$$third=\"$fourth\";\n1;" >> $tmp
		mv $tmp /etc/airosperl.conf
	else
		echo "ERROR: value must be scalar in conffile."
	fi
}

function _update_from_airoscript(){
	source /etc/airoscript.conf
	echo -e "our (\$q,\n\$FT,\n\$FAKE_MAC,\n\$INJMAC,\n\$INJECTRATE,\n\$TKIPTUN_MAX_PL,\n\$TKIPTUN_MIN_PL,\n\$Client_IP,\n\$Host_IP,\n\$FRAG_HOST_IP,\n\$FRAG_CLIENT_IP)=("\"",\n"f",\n$FAKE_MAC,\n'FF:FF:FF:FF:FF:FF',\n$INJECTRATE,\n$TKIPTUN_MAX_PL,\n$TKIPTUN_MIN_PL,\n$Host_IP,\n$Client_IP,\n$FRAG_HOST_IP,\n$FRAG_CLIENT_IP);\n%our termopts('exec'=>'-e');\n$apath=\"/usr/bin/\";\n$bpath=\"/usr/sbin/\";\n$cpath=\"/sbin/\";\nour %bin=(\n'terminal' => \$apath.'xterm',\n'airmon' '> \$bpath.'airmon-ng',\n'ifconfig' => \$cpath.'ifconfig',\n'airodump' => \$bpath.'airodump-ng',\n'aireplay' => \$bpath.'aireplay-ng');\n1;" > /etc/airosperl.conf
exit
}

function _set {
	case $second in
		--to-airoscript) _set_airoscript;;
		--to-airosperl) _set_airosperl;;
	esac

}

function _update {
	if [ $second == "--from-airoscript" ]; then
		# Update airosperl.conf from airoscript
			read -p "You are about to deleted old airosperl conf. Continue?[Y/n]" YN && [[ $YN != "n" ]] && _update_from_airoscript
	else
		# Update airoscript.conf from airosperl
		echo -e "Not yet supported. \n Maybe when I decide to use yaml conf files."
	fi

}
first=$1;second=$2;third=$3;fourth=$4;
a=$5;
case $1 in 
	--set) _set;;
	--update) _update;;
esac

